### PR TITLE
Improve segmentation ICL approach (Maximal Ball method)

### PR DESCRIPTION
**Pull request by volpatto**
## Summary

This PR adds a package-native maximal-ball image-to-network extraction path, introduces a unified network-construction interface across image and imported CNM backends, and expands the verification/docs surface for comparing `voids` against Imperial-style `pnextract` / `pnflow` reference workflows.

## What changed

- add `voids.image.maximal_ball` with staged native maximal-ball extraction utilities, diagnostics, voxel-region assignment, and PoreSpy-style network export
- extend image/network APIs with backend normalization in `extract_spanning_pore_network(...)` and a new `construct_spanning_network(...)` entry point
- add Imperial CNM import support via `voids.io.load_pnflow_cnm(...)`, including optional solver-box compatibility for tighter `pnflow` parity studies
- add benchmark/crosscheck helpers for conduit-conductance auditing and network-geometry summaries/comparisons
- update PoreSpy import/repair handling to support separate throat shape-factor-radius conventions and improved boundary normalization
- add extensive tests for maximal-ball extraction, CNM parsing/import, unified network construction, and geometry/transport crosschecks
- update docs, notebook reports, and example artifacts for image segmentation/network extraction plus DRP-10 and external `pnflow` verification
- add `edt` and `numba` dependencies needed by the native maximal-ball workflow

## Why

- give `voids` a library-native maximal-ball extraction workflow instead of relying only on external extraction behavior
- make backend assumptions explicit and comparable across `porespy`, Imperial-style `snow2`, native maximal-ball extraction, and imported CNM references
- strengthen geometry, boundary-condition, and single-phase verification against saved benchmark cases

## Validation

- `pixi run pytest -q tests/test_maximal_ball.py tests/test_pnflow_cnm_io.py tests/test_network_construction.py tests/test_crosscheck_roundtrip.py tests/test_workflows_porespy_volume.py`
- `pixi run notebooks-smoke`
- `pixi run docs-build`

## Notes

- This branch includes regenerated notebook-report, documentation, and example artifacts alongside code changes.
- I did not rerun the full repository gates while preparing this PR text, so `pixi run test-cov`, `pixi run typecheck`, `pixi run lint`, and `pixi run format-check` are still separate follow-up checks if you want the PR body to claim them explicitly.

### CODE REVIEW

File: src/voids/image/network_extraction.py [hunk 1]
```diff
@@ -253,13 +481,131 @@ def extract_spanning_pore_network(
         net=net,
         pore_indices=pore_indices,
         throat_mask=throat_mask,
-        backend="porespy",
+        backend=backend_normalized,
         backend_version=getattr(ps, "__version__", None),
     )
```
[the remark this answers is not in the document]
Comment (volpatto) [reply src/voids/image/network_extraction.py:491]: @copilot apply changes based on [this feedback](https://github.com/geomech-project/voids/pull/51#discussion_r3198802307)
